### PR TITLE
Example flapping script no render

### PR DESCRIPTION
**Pull request by BlaineEXE**
**Description of your changes:**

Example and not mergeable code. I still left the liveness probes and docs unfinished here, but this is what I mean when we can use the script without rendering it. Notably, looking at commit 2, the pod parsing script doesn't change because the args don't change at all, and the unit test doesn't need updated. The only real change beyond the api config is putting the script in front of the command. That makes it easy to swap the script in/out as needed based on bugs without affecting parsing code.

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/cluster/osd/spec.go [hunk 1]
```diff
@@ -619,7 +635,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 			InitContainers:     initContainers,
 			Containers: []v1.Container{
 				{
-					Command:         osdStartScript(command, args, c.spec.Storage.FlappingRestartIntervalHours),
+					Command:         append([]string{"bash", "-x", "-c", cephOSDStart}, command...),
```
Comment (travisn) [inline pkg/operator/ceph/cluster/osd/spec.go:638]: Thanks for the example of keeping the arguments consistent whether we decide to use the script or not. So supporting both ways of starting the OSD would just still require setting a different command, to either use `bash` or not. 

To support both, a different liveness probe would be needed (one checks for the touched temp file, and one doesn't). But we need to support both liveness probes anyway for other daemons. 

Looking back at #12715, maybe it's not that complicated to support both ways of launching the OSD. @sp98 Can you clarify the complication of looking of the existing properties for the osd info? I think that's not a concern, we could just generate the needed `ceph-osd` or `bash` command in the OSD pod spec depending on the cephcluster CR or operator env var settings. We should need to look at the existing OSD pod to figure out how to generate the spec for this command.

Comment (sp98) [reply pkg/operator/ceph/cluster/osd/spec.go:638]: I tried out this commit as is. If failed to start the OSD. 

```
❯ oc logs rook-ceph-osd-0-54ffb568f-h97pm -n rook-ceph
Defaulted container "osd" out of: osd, log-collector, activate (init), expand-bluefs (init), chown-container-data-dir (init)
+ set -o nounset
+ RESTART_INTERVAL=6
+ child_pid=
+ sigterm_received=false
+ trap sigterm SIGTERM
+ child_pid=530
+ wait 530
+ --foreground --id 0 --fsid 4806bee1-9a4b-42bf-9c31-88f9b77302f3 --setuser ceph --setgroup ceph '--crush-location=root=default host=minikube' --default-log-to-stderr=true --default-err-to-stderr=true --default-mon-cluster-log-to-stderr=true '--default-log-stderr-prefix=debug ' --default-log-to-file=false --default-mon-cluster-log-to-file=false --ms-learn-addr-from-peer=false
ceph-osd: line 15: --foreground: command not found
+ wait 530
+ ceph_osd_rc=127
+ '[' 127 -eq 0 ']'
+ exit 127
```

Comment (BlaineEXE) [reply pkg/operator/ceph/cluster/osd/spec.go:638]: > To support both, a different liveness probe would be needed (one checks for the touched temp file, and one doesn't). But we need to support both liveness probes anyway for other daemons.

I think the liveness probe does its check unless the file is present right? Doesn't that behavior also fit for the method of running osds that doesn't use the script?

File: pkg/operator/ceph/cluster/osd/spec.go [hunk 2]
```diff
@@ -619,7 +635,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 			InitContainers:     initContainers,
 			Containers: []v1.Container{
 				{
-					Command:         osdStartScript(command, args, c.spec.Storage.FlappingRestartIntervalHours),
+					Command:         append([]string{"bash", "-x", "-c", cephOSDStart, "--"}, command...),
```
Comment (BlaineEXE) [inline pkg/operator/ceph/cluster/osd/spec.go:638]: The issue seemed to be the `--` that I added here. That is making it work right in my environment at least. It tells bash definitively to treat the remaining line as arguments to the script.

Comment (sp98) [reply pkg/operator/ceph/cluster/osd/spec.go:638]: This works. Thanks @BlaineEXE

File: pkg/operator/ceph/cluster/osd/spec.go (hunk 1 above)
Comment (sp98) [reply pkg/operator/ceph/cluster/osd/spec.go]: @travisn I tried testing the changes that Blaine has suggested in this PR. Blaine's approach of writing updating the OSD scripts would work and make it easier to fetch the OSDInfo from the POD.  So we can use both the approaches (osd script and directly running osd cmd) in Rook without make a lot of changes. 

We need not have a different liveless probe to support both the scenarios. Liveness probe can always check for the touched temp file whether osd is running as a bash script or just the osd command.

**Pull request closed without merging**